viewing_party/party.py

```diff
@@ -1,23 +1,136 @@
 # ------------- WAVE 1 --------------------
 
+from shutil import move
+
+
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        movie_dict= {}
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+        return movie_dict
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"]== title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+        
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    average_rating = 0
+    sum = 0
+    lenght = len(user_data["watched"])
+    for i in range(lenght):
+        sum += user_data["watched"][i]["rating"]
+        average_rating = sum/lenght 
+    return average_rating

```

Comment:
This works well! One small thing: Consider using a for-each instead of a for loop here. The value of the index is not important.